packages/webdriverio/tests/middleware.test.js

```diff
@@ -0,0 +1,61 @@
+import logger from '@wdio/logger'
+import { remote } from '../src'
+import request from 'request'
+
+const { warn  } = logger()
+const waitForExist = require('../src/commands/element/waitForExist')
+
+describe('middleware', () => {
+    let browser;
+
+    beforeAll( async () => {
+        browser = await remote({
+            baseUrl: 'http://foobar.com',
+            capabilities: {
+                browserName: 'foobar'
+            },
+            waitforInterval: 20,
+            waitforTimeout: 100
+        })
+    })
+
+    it('should throw an error if the element is never found', async () => {
+        waitForExist.default =
+            jest.fn().mockImplementation ( ( ) => { throw new Error(`Promise was rejected with the following reason`)});
+
+        const elem = await browser.$('#foo')
+        elem.elementId = undefined
+
+        await expect(elem.click())
+            .rejects.toThrow(`Can't call click on element with selector "#foo" because element wasn't found`)
+    });
+
+    it('should succesfully click on an element that falls stale after being refound', async () => {
+        waitForExist.default =
+            jest.fn().mockImplementation (( ) => { return true });
+
+        const elem = await browser.$('#foo')
+        const subElem = await elem.$('#subfoo')
+        const subSubElem = await subElem.$('#subsubfoo');
```

Comment:
this should fail the lint test because the project doesn't do `;`

Reply:
It might be better to add linter checks to CI for avoiding spending time on it during review

Reply:
@CrispusDH yes, I was assuming that eslint is already configured this way but was wrong about that. So @abjerstedt please ignore this, I will make a separate PR with a lint rule patch and fixing all semicolon appearances

Reply:
@CrispusDH There are. We were discussing this in the gitter channel, and it seems the rule isn't strictly enforced. Christian said we can ignore this comment for now, and enforce in a follow-up pr.